lib/rubocop/cop/style/redundant_parentheses.rb

```diff
@@ -177,6 +180,12 @@ def find_offense_message(begin_node, node)
         # @!method interpolation?(node)
         def_node_matcher :interpolation?, '[^begin ^^dstr]'
 
+        def argument_of_parenthesized_method_call?(node)
+          return false unless (parent = node.parent)
+
+          parent.call_type? && parent.parenthesized? && parent.receiver != node
+        end
```

Comment:
This currently doesn't catch parenthesized keyword arguments, e.g.:
```ruby
foo(bar: (baz + qux))
```

Let me know if this makes sense to be an offense and if I should add this logic in the same PR. I'm on the fence about this because of some examples I saw online, e.g. [in Gitlab codebase](https://gitlab.com/gitlab-org/gitlab/-/blob/82d5de42558d34b6a57ff4597035b9e8fff147ef/lib/gitlab/gitaly_client/commit_service.rb#L411-412):
```ruby
method(
  range: (encode_binary(range) if range),
  ignore_revisions_blob: (encode_binary(ignore_revisions_blob) if ignore_revisions_blob)
)
```

Reply:
These can be considered for a separate update later. For now, the changes in this PR seem reasonable.